h2o-core/src/test/java/water/fvec/VecTest.java

```diff
@@ -148,4 +146,50 @@ public void map(Chunk c) {
     source.remove(new Futures()).blockForPending();
     con.remove(new Futures()).blockForPending();
   }
+
+  @Test public void testChunkForChunkIdxAfterVecUpdate() {
+    try {
+      Scope.enter();
+      Vec v1 = Scope.track(makeSeq(2 * FileVec.DFLT_CHUNK_SIZE, false));
+      Vec v2 = Scope.track(makeSeq(2 * FileVec.DFLT_CHUNK_SIZE, false));
+      Chunk c0 = v1.chunkForChunkIdx(0);
+      c0._vec = v2; // inject any vec into the cached POJO, this simulates a stale Vec reference
+      Chunk c0c = v1.chunkForChunkIdx(0);
+      assertSame(c0, c0c);
+      // the vec reference was updated in the existing live object
+      assertSame(v1, c0c._vec);
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test public void testChunkForChunkIdxMRTask() {
+    Assume.assumeTrue(H2O.getCloudSize() > 0);
```

Comment:
This is always true. `> 1` perhaps ?

Reply:
Yes!!! That was my intention, will fix. Thanks for noticing. The test doesn't test much on a single node :)